staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/envelope/vault/client.go

```diff
@@ -0,0 +1,212 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package vault implements envelop encryption provider based on Vault KMS
+package vault
+
+import (
+	"fmt"
+	"path"
+	"reflect"
+	"strings"
+
+	"github.com/hashicorp/vault/api"
+)
+
+// Handle all communication with Vault server.
+type clientWrapper struct {
+	client      *api.Client
+	encryptPath string
+	decryptPath string
+	authPath    string
+}
+
+// Initialize a client wrapper for vault kms provider.
+func newClientWrapper(config *EnvelopeConfig) (*clientWrapper, error) {
+	client, err := newVaultClient(config)
+	if err != nil {
+		return nil, err
+	}
+
+	// Vault transit path is configurable. "path", "/path", "path/" and "/path/"
+	// are the same.
+	transit := "transit"
+	if config.TransitPath != "" {
+		transit = strings.Trim(config.TransitPath, "/")
+	}
+
+	// auth path is configurable. "path", "/path", "path/" and "/path/" are the same.
+	auth := "auth"
+	if config.AuthPath != "" {
+		auth = strings.Trim(config.AuthPath, "/")
+	}
+	wrapper := &clientWrapper{
+		client:      client,
+		encryptPath: "/" + path.Join("v1", transit, "encrypt") + "/",
+		decryptPath: "/" + path.Join("v1", transit, "decrypt") + "/",
+		authPath:    "/" + path.Join(auth) + "/",
```

Comment:
You don't need to prefix or suffix this with slashes it can just be `path.Join(stuff...)`

Reply:
fixed